src/gui/owncloudsetupwizard.cpp

```diff
@@ -167,18 +167,22 @@ void OwncloudSetupWizard::slotFoundServer(const QUrl &url, const QJsonObject &in
         if (url.scheme() == QLatin1String("https") && oldUrl.host() == url.host()) {
             _ocWizard->account()->setUrl(url);
         } else {
-            auto accountState = new AccountState(_ocWizard->account());
-            connect(accountState, &AccountState::urlUpdated, this, [accountState, this] {
-                accountState->deleteLater();
-                slotDetermineAuthType();
-            });
-            accountState->updateUrlDialog(url);
+            auto account = _ocWizard->account();
+            connect(account.data(), &Account::urlChanged, this, &OwncloudSetupWizard::slotUrlUpdated);
+            AccountState::updateUrlDialog(account, url);
             return;
         }
     }
     slotDetermineAuthType();
 }
 
+void OwncloudSetupWizard::slotUrlUpdated()
+{
+    auto account = qobject_cast<Account *>(sender());
+    disconnect(account, &Account::urlChanged, this, &OwncloudSetupWizard::slotUrlUpdated);
```

Comment:
I prefer the lambda for small connects like that, moving the 3 line to a function doesn't help with the readability at all.